docs/proposals/hosted_components_templates.md

```diff
@@ -0,0 +1,554 @@
+# Use [OCP templates and] ABPs to install hosted components
+
+## Description
+Currently most of our hosted components have their own repositories where they
+maintain changes and then need to synchronize those changes within their
+respective roles. This can cause cause delays and times where features don't
+match up with what is installed or can be configured.
+
+There is also a concern that copying files from one location to another can
+cause cases where files are out of sync or versions don't align with what is
+installed and what is configured (is what we configured correct for the image
+version we installed).
+
+## Rationale
+As was piloted with the Template Service Broker, a role was created to process
+and configure OCP templates that were synchronized (currently still a manual
+process) from the origin repo where the component resides.
+
+This allowed any changes for the installation to be maintained by the developers
+and keep them from needing to be fluent with Ansible. It also afforded a level
+of abstraction from needing to keep in sync with TSB related changes and then
+translate them to changes within an Ansible role.
+
+However, there is still the gap of ensuring our role configures changes appropriate
+for the version of the component we are installing. Prior to 3.4 with Aggregated
+Logging this was resolved and contained due to using a deployer pod, since everything
+was versioned and contained within an artifact the Integration Services team
+provided. This is something that is addressed again now with the Service Broker,
```

Comment:
Is "Service Broker" the same as "Template Service Broker", or "TSB". If so, can we just use "TSB"?

Reply:
sort of... a Service Broker is more generic and the TSB is a specific one... there can also be the Ansible Service Broker